amd/src/app/block/element.js

```diff
@@ -170,6 +170,70 @@ export default class BlockElement {
         const bulkDeleteButton = this.#element.querySelector('#block_sharing_cart_bulk_delete_confirm');
 
         const checkboxSelector = '.sharing_cart_item input[data-action="bulk_select"][type="checkbox"]';
+        const selectAllCheckbox = this.#element.querySelector('#select_all_box');
+        const selectAllLabel = this.#element.querySelector('#select_all_label');
+
+        const cancelBulkDeleteButton = this.#element.querySelector('#block_sharing_cart_cancel_bulk_delete');
+        const bulkDeleteTrigger = this.#element.querySelector('#block_sharing_cart_bulk_delete');
+
+
+
+        const getItemCheckboxes = () => this.#element.querySelectorAll(checkboxSelector);
+
+        const updateBulkDeleteButtonState = () => {
+            bulkDeleteButton.disabled = !Array.from(getItemCheckboxes()).some(checkbox => checkbox.checked);
+        };
+
+        const updateSelectAllState = async () => {
+            const itemCheckboxes = getItemCheckboxes();
+            const allSelected = Array.from(itemCheckboxes).every(checkbox => checkbox.checked);
+            const someSelected = Array.from(itemCheckboxes).some(checkbox => checkbox.checked);
+            selectAllCheckbox.checked = allSelected;
+            selectAllCheckbox.indeterminate = !allSelected && someSelected;
+            selectAllLabel.textContent = allSelected ?
+                await get_string('deselect_all', 'block_sharing_cart') :
+                await get_string('select_all', 'block_sharing_cart');
+        };
+
+        const toggleSelectAll = () => {
+            const itemCheckboxes = getItemCheckboxes();
+            const allSelected = Array.from(itemCheckboxes).every(checkbox => checkbox.checked);
+            itemCheckboxes.forEach(checkbox => {
+                checkbox.checked = !allSelected;
+            });
+            updateBulkDeleteButtonState();
+            updateSelectAllState();
+        };
```

Comment:
Can just be moved down where it's used since it's only used once, then we can also avoid saving it as a const.
```suggestion
```